ui/site/src/site.ts

```diff
@@ -26,6 +26,11 @@ lichess.info = info;
 lichess.load.then(() => {
   $('#user_tag').removeAttr('href');
 
+  let puzzleDifficulty = lichess.storage.get('analyse.puzzle.difficulty') ?? 'normal';
+  if (puzzleDifficulty !== 'normal') {
+    $("a[href='/training']").attr('href', `/training?difficulty=${puzzleDifficulty}`);
+  }
+
```

Comment:
I don't understand the idea of putting the difficulty in the puzzle URL. I also don't think that hacking into the top menu, on every page, to change the puzzle URL is a good idea. It's inefficient, almost always unused, and only handles the use case where puzzles are accessed through that link. But there are other ways to get there.